src/ImageCommand.cc

```diff
@@ -235,85 +235,90 @@ Json::Value FindImage::construct_responses(
 
     Json::Value& findImage = responses[0];
 
-    assert(findImage.isMember("entities"));
-
     if (findImage["status"] != 0) {
         findImage["status"]  = RSCommand::Error;
         // Uses PMGD info error.
         return error(findImage);
     }
 
-    bool flag_empty = true;
+    Json::Value results = get_value<Json::Value>(cmd, "results");
```

Comment:
Actually I am quite confused with this commit. Shouldn't we check in construct_protobuf if blob == false and no list/count/average requested? Or is this relying on PMGDQueryHandler throwing an error and then handling it construct_responses? Ideally we should put it in JSON checks but I am guessing that's not possible. What am I missing?